tests/sentry/integrations/github/test_client.py

```diff
@@ -746,3 +749,579 @@ def test_a_slow_integration_is_broken(self):
         with pytest.raises(Exception):
             self.gh_client.get_blame_for_file(self.repo, "foo.py", "main", 1)
         assert Integration.objects.get(id=self.integration.id).status == ObjectStatus.DISABLED
+
+
+class GitHubClientFileBlameQueryBuilderTest(TestCase):
+    """
+    Tests that get_blame_for_files builds the correct GraphQL query
+    """
+
+    @mock.patch("sentry.integrations.github.client.get_jwt", return_value=b"jwt_token_1")
+    def setUp(self, get_jwt):
+        integration = self.create_integration(
+            organization=self.organization,
+            provider="github",
+            name="Github Test Org",
+            external_id="1",
+            metadata={"access_token": None, "expires_at": None},
+        )
+        self.repo_1 = Repository.objects.create(
+            organization_id=self.organization.id,
+            name="Test-Organization/foo",
+            url="https://github.com/Test-Organization/foo",
+            provider="integrations:github",
+            external_id=123,
+            integration_id=integration.id,
+        )
+        self.repo_2 = Repository.objects.create(
+            organization_id=self.organization.id,
+            name="Test-Organization/bar",
+            url="https://github.com/Test-Organization/bar",
+            provider="integrations:github",
+            external_id=456,
+            integration_id=integration.id,
+        )
+        install = integration.get_installation(organization_id=self.organization.id)
+        assert isinstance(install, GitHubIntegration)
+        self.install = install
+        self.github_client = self.install.get_client()
+        responses.add(
+            method=responses.POST,
+            url="https://api.github.com/app/installations/1/access_tokens",
+            body='{"token": "12345token", "expires_at": "2030-01-01T00:00:00Z"}',
+            status=200,
+            content_type="application/json",
+        )
+
+    @mock.patch("sentry.integrations.github.client.get_jwt", return_value=b"jwt_token_1")
+    @responses.activate
+    def test_get_blame_for_files_same_repo(self, get_jwt):
+        """
+        When all files are in the same repo, only one repository object should be
+        queried, but files blames within the repo should be deduped
+        """
+        file1 = SourceLineInfo(
+            path="src/sentry/integrations/github/client_1.py",
+            lineno=10,
+            ref="master",
+            repo=self.repo_1,
+            code_mapping=None,  # type: ignore
```

Comment:
In this case, I wanted to pass None for `code_mapping ` to make the test simpler. `code_mapping` is a required parameter, but doesn't affect the logic of the function, so I thought it would be fine to have this override in the test. But if you disagree I'll add it into the test properly